src/Core/Resolvers/Sql Query Structures/BaseSqlQueryStructure.cs

```diff
@@ -143,8 +143,85 @@ public Type GetColumnSystemType(string columnName)
         /// <summary>
         /// Based on the relationship metadata involving referenced and
         /// referencing columns of a foreign key, add the join predicates
-        /// to the subquery Query structure created for the given target entity Name
-        /// and related source alias.
+        /// to the subquery QueryStructure created for the given target's entity name
+        /// and related target table source alias (name of table used in subquery).
+        /// There are only a couple of options for the foreign key - we only use the
+        /// valid foreign key definition. It is guaranteed at least one fk definition
+        /// will be valid since the MetadataProvider.ValidateAllFkHaveBeenInferred.
+        /// </summary>
+        /// <param name="fkLookupKey">{entityName, relationshipName} used to lookup foreign key metadata.</param>
+        /// <param name="targetEntityName">Related(target) entity's name.</param>
+        /// <param name="subqueryTargetTableAlias">The alias assigned for the underlying source of this related entity.</param>
+        /// <param name="subQuery">The subquery to which the join predicates are to be added.</param>
+        public void AddJoinPredicatesForRelationship(
+            EntityRelationshipKey fkLookupKey,
+            string targetEntityName,
+            string subqueryTargetTableAlias,
+            BaseSqlQueryStructure subQuery)
+        {
+            if (string.Equals(fkLookupKey.EntityName, targetEntityName))
+            {
+                AddJoinPredicatesForSelfJoinedEntity(
+                    fkLookupKey: fkLookupKey,
```

Comment:
Is this fkLookupKey is not useful today when its not a self joined relationship? If we start using it for relationships between different entities, can we for e.g. solve the problem of having multiple relationships between same pair of different entities?

Reply:
With this PR, `fkLookupKey` is only useful for self-joined relationships. I initially authored this PR to utilize the `fkLookupKey` for all types of relationships, but more issues came up which required even more cascading changes make work. As a result, I'm introducing these changes incrementally. First -> to enable self-join relationships. In the future, can adopt this new way of persisting relationshipName-> fkMetadata to non-self-join relationships.

Reply:
Is there an issue tracking this work to allow multiple relationships between same pair of different entities? 

Also, will this PR solve this issue as well: https://github.com/Azure/data-api-builder/issues/1718 ?

Reply:
By:
> this work to allow multiple relationships between same pair of different entities?

do you mean like the following? I'm not seeing how that is different from #1718:

```json
{
  "entities" {
    "MyEntity" {
      "relationships": {
         "Relationship1": {
             "targetEntity": "YourEntity", // 
             "sourceCols": ["col1"],
             "targetCols": [ "colA"]
           },
         "Relationship2" {
             "targetEntity": "YourEntity",
             "sourceCols": ["col2"],
             "targetCols": [ "colb"]
}}

Reply:
1718 is not fixed via this change.